File: lucette/src/main/java/battle/csn/lucette/machine/states/ErrorState.java
package battle.csn.lucette.machine.states;

import org.apache.log4j.Logger;

import battle.csn.lucette.machine.Chain;
import battle.csn.lucette.machine.states.framework.AbstractFinalState;


public class ErrorState extends AbstractFinalState
{
    private static final Logger LOGGER = Logger.getLogger( ErrorState.class );
    private String message;

    public ErrorState( String message )
    {
        this.message = message;
    }

    @Override
    public void pull( Chain wrapper )
    {
        LOGGER.warn( "ERREUR : " + message );
    }

}

File: trunk/lucette2/src/main/scala/battle/csn/lucette2/engine/states/ErrorState.scala
package battle.csn.lucette2.engine.states

import battle.csn.lucette2.engine.states.framework.AbstractFinalState
import battle.csn.lucette2.engine.Chain
import org.apache.log4j.Logger

class ErrorState(message: String) extends AbstractFinalState {
  def LOGGER = Logger.getLogger(classOf[ErrorState]);

  override def pull(wrapper: Chain) {
    LOGGER.error("ERREUR : " + message)
  }
}